connect/api/src/main/java/org/apache/kafka/connect/data/Struct.java

```diff
@@ -229,7 +229,12 @@ public void validate() {
             Object value = values[field.index()];
             if (value == null && (fieldSchema.isOptional() || fieldSchema.defaultValue() != null))
                 continue;
-            ConnectSchema.validateValue(fieldSchema, value);
+            try {
+                ConnectSchema.validateValue(fieldSchema, value);
+            } catch(DataException e) {
+                throw new DataException("Validate failed for required field: \""  + field.name() + "\", "
```

Comment:
Strictly speaking you don't know whether this is a required field here. Also, you probably don't need to print `e.getMessage()` again since the exception have already included this message by including `e` in the newly constructed exception.

Actually, if all you need is the field name, would it be simpler to just keep the Struct.java as it is, and change ConnectSchema.java to do `throw new DataException("Invalid value: null used for required field " + schema.name())`